pkg/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -233,39 +205,46 @@ func createLogicalReplicationStreamPlanHook(
 		}
 		crossClusterResolver := crosscluster.MakeCrossClusterTypeResolver(sourceTypes)
 
+		replicationStartTime := spec.ReplicationStartTime
+		progress := jobspb.LogicalReplicationProgress{}
+		if cursor, ok := options.GetCursor(); ok {
+			replicationStartTime = cursor
+			progress.ReplicatedTime = cursor
+		}
+
 		// If the user asked to ignore "ttl-deletes", make sure that at least one of
 		// the source tables actually has a TTL job which sets the omit bit that
 		// is used for filtering; if not, they probably forgot that step.
 		throwNoTTLWithCDCIgnoreError := discard == jobspb.LogicalReplicationDetails_DiscardCDCIgnoredTTLDeletes
 
+		// TODO: consider moving repPair construction into doLDRPlan after dest tables are created.
```

Comment:
@dt before i do this additional refactor, i'm curious what you think of the patch as it stands now.